app/models/invite_code_validator.rb

```diff
@@ -0,0 +1,29 @@
+class InviteCodeValidator < ActiveModel::Validator
+  def validate(user)
```

Comment:
> Shouldn't they be the same? 
Only if the invite code actually exists. If it does not exist user.invite_code can be anything - and user_invite_code will be nil.
So if you fetch the invite_code and then hand the record to `not_existent?` you will only need to check for `.nil?` - no need to try and load the record another time.

> (I don't think we can use user.invite_code for both because used_already? needs the invite_count, and user doesn't have one.)
Right. So the fetching of user_invite_code from line 4 would have to move into the used_already? function. We're only using it in that function anyway.